fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -0,0 +1,226 @@
+     * <p>
+     * The return value is a map of duplicate <tt>MetricColumn</tt> to all <tt>Availability</tt>'s that contains
+     * this <tt>MetricColumn</tt>
+     * <p>
+     * For example, when a map of {availability1: [metricCol1, metricCol2], availability2: [metricCol1]} is passed to
+     * this method, the method returns {metricCol1: [availability1, availability2]}
+     *
+     * @param availabilityToAvailableColumns  A map from <tt>Availability</tt> to set of <tt>MetricColumn</tt>
+     * contained in that <tt>Availability</tt>
+     *
+     * @return duplicate values of <tt>MetricColumn</tt>'s
+     */
+    private static Map<MetricColumn, Set<Availability>> getDuplicateValue(
+            Map<Availability, Set<MetricColumn>> availabilityToAvailableColumns
+    ) {
+        Map<MetricColumn, Set<Availability>> metricColumnSetMap = new HashMap<>();
+
+        for (Map.Entry<Availability, Set<MetricColumn>> entry : availabilityToAvailableColumns.entrySet()) {
+            Availability availability = entry.getKey();
+            for (MetricColumn metricColumn : entry.getValue()) {
+                metricColumnSetMap.put(
+                        metricColumn,
+                        Sets.union(
+                                metricColumnSetMap.getOrDefault(metricColumn, new HashSet<>()),
+                                Sets.newHashSet(availability)
+                        )
+                );
+            }
+        }
+
+        // For example, when a map of {availability1: [metricCol1, metricCol2], availability2: [metricCol1]} is
+        // passed to this method, at this point,
+        // "metricColumnSetMap" =  {metricCol1: [availability1, availability2], {metricCol2: [availability1]}}
+        // The duplicate metric columns is "metricCol1" which can be selected by knowing that the value of "metricCol1"
+        // has a collection whose size is greater than 1; with that, we return
+        // {metricCol1: [availability1, availability2]}
+        return metricColumnSetMap.entrySet().stream()
+                .filter(entry -> entry.getValue().size() > 1)
+                .collect(
+                        Collectors.toMap(
+                                Map.Entry::getKey,
+                                Map.Entry::getValue
+                        )
+                );
+    }
+
+    /**
+     * Given a <tt>DataSourceConstraint</tt> - DSC1, construct a mapping from a type of <tt>Availability</tt>, A, to its
+     * <tt>DataSourceConstraint</tt>, DSC2. DSC2 is constructed as the intersection of metric columns between DSC1 and
+     * A's available metric columns. There are cases in which the intersection is empty; this method filters out
+     * mapping entries that maps to <tt>DataSourceConstraint</tt> with empty set of metric names.
+     *
+     * @param constraint data constraint whose contained metric columns will be intersected with availabilities'
+     * metric columns
+     *
```

Comment:
spacing, two spaces after param name. 
Use an article on param description, e.g.: 
@param constraint 'the data constraint' or 'a data constraint'.